src/hotspot/share/opto/superword.cpp

```diff
@@ -3615,3 +3029,279 @@ bool SuperWord::same_origin_idx(Node* a, Node* b) const {
 bool SuperWord::same_generation(Node* a, Node* b) const {
   return a != nullptr && b != nullptr && _clone_map.same_gen(a->_idx, b->_idx);
 }
+
+void SuperWordVTransformBuilder::build_vtransform() {
+  assert(!_packset.is_empty(), "must have non-empty packset");
+
+  build_vector_vtnodes_for_packed_nodes();
+  build_scalar_vtnodes_for_non_packed_nodes();
+
+  VectorSet vtn_dependencies; // Shared, but cleared for every vtnode.
+  build_edges_for_vector_vtnodes(vtn_dependencies);
+  build_edges_for_scalar_vtnodes(vtn_dependencies);
```

Comment:
This is a little confusing since you are not only adding edges but also new vtnodes. Maybe you find a better name. But I suggest to at least add some comments at these methods to mention that. Otherwise, it reads like: Build vector vtnodes, then scalar vtnodes, then connect them somehow (without new nodes being added).

Reply:
That's better, I like it! Before that it first suggested (without looking at the code in more detail) that we have a fixed set of vtnodes and then just build some edges (now it's even more clear that it's about inputs and not outputs).